Auto-save

ebolam opened the pull request:
This code change adds a toggle switch to the settings menu to enable auto-save. When enabled, any time the action list is changed it triggers a save.

vfbd commented:
Can you make it so that the auto-saving only triggers if you're using a loaded save? If you were to edit a story that has no corresponding save file, the save dialog would pop up repeatedly.

henk717 commented:
If you can please submit this to the united branch on my own github since its the next release and nearly complete.

ebolam commented:
Reworked to fix the save bug on new stories (now waits for the filename to have .json in it before allowing an autosave). Also moved the pull request to the united branch (https://github.com/henk717/KoboldAI/pull/67)
